### ISSUE TITLE
Any plan to make the stats component extensible?

### ISSUE THREAD

**Issue by t-bzhan** [Azure/durabletask#423]
It seems that AzureStorageOrchestrationService use AzureStorageOrchestrationServiceStats instead of some general interface to track internal metrics. That makes it is not extensible and hard to export the metrics to third party monitoring platform like Prometheus, Azure Monitor etc.
https://github.com/Azure/durabletask/blob/66e19e1262d6a29949afe9390f7556a8b44f5a63/src/DurableTask.AzureStorage/Monitoring/AzureStorageOrchestrationServiceStats.cs#L19-L38

Though one could listen to the "DurableTask-AzureStorage" event source and parse the log to get the metrics, but that did not sound very elegant.

Do we have any plan to improve it?

**Comment by cgillum** [comment]
Yes, we’re planning on making all tracing extensible, including these stats by also writing the event source traces to an ILogger interface. This would allow you to export these logs easily. You simply have to provide the worker with an ILoggerFactory. It’s under active development and I should have a PR out syndrome writing the next couple weeks. Let me know if you think something like this would work for you.

**Comment by t-bzhan** [comment]
@cgillum , thanks so much, that would be great.
